cloud/azure/bin/deploy

```diff
@@ -1,56 +1,37 @@
 #! /usr/bin/env bash
 set -e
 
-source "cloud/shared/bin/lib/health.sh"
+source "cloud/shared/bin/lib.sh"
 source "cloud/azure/bin/lib.sh"
 
+# loads and validates config variables and generates tfvars file
 python3 "cloud/shared/bin/lib/tf_apply_setup.py"
-readonly TF_VAR_FILE="setup.auto.tfvars"
-readonly BACKEND_VARS_FILENAME="backend_vars"
-
-cloud/azure/bin/setup_tf_shared_state "${TERRAFORM_TEMPLATE_DIR}/${BACKEND_VARS_FILENAME}"
-
-readonly USER_ID="$(azure::get_current_user_id)"
 
+azure::set_common_vars
 azure_log::fetch_log_file
 
-set +e
-(
-    set -e
-    terraform \
-        -chdir="${TERRAFORM_TEMPLATE_DIR}" \
-        init \
-        -backend-config="${BACKEND_VARS_FILENAME}"
-
-    terraform \
-        -chdir="${TERRAFORM_TEMPLATE_DIR}" \
-        apply \
-        -var-file="${TF_VAR_FILE}"
-)
-readonly TF_STATUS=$?
-set -e
-
-if (( $TF_STATUS != 0 )); then
-    log::deploy_failed "${IMAGE_TAG}" "${USER_ID}"
+if ! terraform::perform_apply; then
```

Comment:
slightly prefer to have this live alone and then store as a var and then check the result of it

Reply:
like this?
```suggestion
set +e
terraform::perform_apply
readonly TF_STATUS=$?
set -e

if (( $TF_STATUS != 0 )); then
...
```

Reply:
ugh no the way you had it was much better